sefaria/model/text.py

```diff
@@ -1395,19 +1395,26 @@ def version(self):
 
     def text_index_map(self,tokenizer):
```

Comment:
We need some tests for this, depth 1,2, & 3, with variations of spacing, paragraphs, etc.